x-pack/plugins/observability_solution/synthetics/server/common/inline_to_zip.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import archiver from 'archiver';
+import { MemWritable } from './mem_writable';
+
+function wrapInlineInProject(inlineJourney: string) {
+  return `import { journey, step, expect } from '@elastic/synthetics';
+
+journey('inline', ({ page, context, browser, params, request }) => {
+${inlineJourney}
+});`;
+}
+
+export async function inlineToProjectZip(
+  inlineJourney: string,
+  monitorId: string,
+  logger: Logger
+): Promise<string> {
+  const mWriter = new MemWritable();
+  try {
+    await new Promise<void>((resolve, reject) => {
+      const archive = archiver('zip', {
+        zlib: { level: 9 },
+      });
+      archive.on('error', reject);
+      mWriter.on('close', resolve);
+      archive.pipe(mWriter);
+      archive.append(wrapInlineInProject(inlineJourney), {
+        name: 'inline.journey.ts',
```

Comment:
can you throw an error and see the stack trace? I would like us to avoid capturing the Kibana stack trace with this change. We should only show the `inline.journey.ts`

Reply:
Server log looks like:

```
[2024-10-16T15:24:18.403-04:00][ERROR][plugins.synthetics] Error: fail to zip
    at wrapInlineInProject (inline_to_zip.ts:13:9)
    at inline_to_zip.ts:35:22
    at new Promise (<anonymous>)
    at inlineToProjectZip (inline_to_zip.ts:28:11)
    at mapInlineToProjectFields (map_inline_to_project_fields.ts:31:48)
    at handler (run_once_monitor.ts:57:63)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

Reply:
Can you test when you throw error inside the inline monitors? 
```ts
step( 'error', () => {
  throw new Error('boom');
});
```

Reply:
@vigneshshanmugam does the stacktrace looks good?

Reply:
Apologies for the delay. LGTM.